File: silver/1940.py
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())
nums = list(map(int, input().split()))
nums.sort(reverse = True)

cnt = 0
i = 0
j = n - 1
while i < j:
    if nums[i] + nums[j] == m:
        i += 1
        j -= 1
        cnt += 1
    elif nums[i] + nums[j] > m:
        i += 1
    else:
        j -= 1        
            
print(cnt)

File: silver/1940.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	int n, m;
	cin >> n >> m;

	vector<int> ingred(n);
	for (int i = 0; i < n; i++)
		cin >> ingred[i];
	sort(ingred.begin(), ingred.end());

	int s=0, e=n-1, cnt = 0;
	while (s < e) {
		if (ingred[s] + ingred[e] == m){
			cnt++;
			e--;
			s++;
		}
		else if (ingred[s] + ingred[e] > m){
			e--;
		}
		else
			s++;

	}
	cout << cnt;
}